deploy/lib/createFunctions.js

```diff
@@ -15,9 +15,42 @@ module.exports = {
     const { functions } = this.provider.serverless.service;
 
     const functionNames = Object.keys(functions);
+
+    const deletedFunctionsNames = [];
+
+    if (this.serverless.configurationInput.singleSource !== undefined
+      && this.serverless.configurationInput.singleSource !== null
+      && this.serverless.configurationInput.singleSource === true) {
+      // if a function is available in the API but not in the serverlsssyml file, remove it
+      for (let i = 0; i < foundFunctions.length; i++) {
+        const apiFunc = foundFunctions[i];
+
+        for (let ii = 0; ii < functionNames.length; ii++) {
+          const funcName = functionNames[ii];
+
+          if (apiFunc === funcName) {
+            functionNames.slice(ii, 1);
+            break;
+          }
+        }
```

Comment:
There's a lot of shared logic between the function and container versions of this operation. Can we factor this filtering of functions into a utility function somewhere?

Reply:
fixed